http/handler.go

```diff
@@ -42,15 +45,18 @@ func respondError(w http.ResponseWriter, status int, err error) {
 	json.NewEncoder(w).Encode(resp)
 }
 
-func respondOk(w http.ResponseWriter, body interface{}) {
+func respondOk(ctx context.Context, w http.ResponseWriter, body interface{}) {
 	w.Header().Add("Content-Type", "application/json")
 
-	if body == nil {
-		w.WriteHeader(http.StatusNoContent)
-	} else {
-		w.WriteHeader(http.StatusOK)
+	status := http.StatusNoContent
+
+	if body != nil {
+		status = http.StatusOK
 		json.NewEncoder(w).Encode(body)
 	}
+
+	w.WriteHeader(status)
```

Comment:
`w.WriteHeader` needs to go before you write anything to `w` (which `json.NewEncoder(w).Encode(body)` does), so I don't think will work :/